### DISCUSSION TITLE
Any way to specify what not to update in an upsert?

### DISCUSSION THREAD

**Discussion by omer-oa1904193** [mikro-orm/mikro-orm#4317] [Q&A]
Sometimes I need an upsert that keeps some of the fieldsas is if the row already exists but inserts a new value for them if it doesn't. In an SQL query that is simple, just omit the field from the do update part:

```SQL
insert into "my_table" ("field1", "field2", "field3", "field4")
values ('value1', 'value2', 'value3', 'value4')
on conflict ("field1") do update set "field2" = excluded."field2",
                                     "field3" = excluded."field3"
returning "id"
```

Here `'value4'` will be inserted for `field4` if there's no conflict, but if there is conflict `field4` won't be changed. 

Is there a way to do this with the ORM's `em.upsertMany` function? I see all the fields other than the unique ones are included in the update clause. Any way to exlude some of them from the update but keep in the insert?

If not I think there should be.

**Comment by B4nan** [comment]
You can use query builder to have absolute control. I don't mind making things configurable in the EM API, so PR welcome for that.

**Comment by squivix** [comment]
I'll try to work on it
